### PR TITLE
Bug/em 2159 enhanced textlayer highlighting

### PR DESCRIPTION
**Pull request by velimd**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/EM-2159

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: projects/media-viewer/src/lib/annotations/annotation-set/annotation-set.component.ts [hunk 1]
```diff
@@ -127,14 +127,24 @@ export class AnnotationSetComponent implements OnInit, OnDestroy {
   async createTextHighlight(highlight: Highlight) {
     if (highlight.page === this.page) {
       if (window.getSelection()) {
+        const localElement = (<HTMLElement>highlight.event.target) || (<HTMLElement>highlight.event.srcElement);
```
Comment (linusnorton) [inline projects/media-viewer/src/lib/annotations/annotation-set/annotation-set.component.ts]: Maybe: 

```
const localElement: HTMLElement = highlight.event.target || highlight.event.srcElement;
```

File: projects/media-viewer/src/lib/annotations/annotation-set/annotation-set.component.ts [hunk 2]
```diff
@@ -127,14 +127,24 @@ export class AnnotationSetComponent implements OnInit, OnDestroy {
...
+
+        if (localElement.parentElement.childNodes) {
+          localElement.parentElement.childNodes.forEach(child => {
+            child['style']['padding'] = 0;
+            // regex will be targeting the translate style in string
+            // e.g. scaleX(0.969918) translateX(-110.684px) translateY(-105.274px) will become scaleX(0.969918)
+            const translateCSSRegex = /translate[XYZ]\(-?\d*(\.\d+)?(px)?\)/g;
+            child['style']['transform'] = child['style']['transform'].replace(translateCSSRegex, '');
+          });
+        }
```
Comment (linusnorton) [inline projects/media-viewer/src/lib/annotations/annotation-set/annotation-set.component.ts]: I would just do:

```
for (const child of localElement.parentElement.childNodes || []) {
  // ...
}
```

File: projects/media-viewer/src/lib/annotations/annotation-set/annotation-set.component.ts (hunk 1 above)
Comment (velimd) [reply projects/media-viewer/src/lib/annotations/annotation-set/annotation-set.component.ts]: this is not possible, as it doesn't covert it into HTMLElement, it just rejects it.

File: projects/media-viewer/src/lib/annotations/annotation-set/annotation-set.component.ts (hunk 2 above)
Comment (velimd) [reply projects/media-viewer/src/lib/annotations/annotation-set/annotation-set.component.ts]: Type 'NodeListOf<ChildNode>' must have a '[Symbol.iterator]()' method that returns an iterator.

Comment (linusnorton) [reply projects/media-viewer/src/lib/annotations/annotation-set/annotation-set.component.ts]: :man_facepalming: their array like object doesn't implement Symbol.iterator

**Pull request merged**